src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -852,8 +862,16 @@ def _prepare_deployment_properties_unmodified(cmd, template_file=None, template_
     template_obj = None
     on_error_deployment = None
     template_content = None
+
+    if query_string and not template_uri:
+        raise IncorrectUsageError('please provide --template-uri if --query-string is specified')
```

Comment:
Do we need to consider adding some verifications for `query_string` to make sure it is vaild?

Reply:
This is done on the server side.